src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/BuildPropertyPage.VisualBasic.xaml

```diff
@@ -1,6 +1,375 @@
+
+  <!-- TODO: Replace HelpUrl with a fwlink -->
+  <BoolProperty Name="Optimize"
+                DisplayName="Enable optimizations"
+                HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/advanced-compiler-settings-dialog-box-visual-basic?view=#optimizations"
+                Category="Advanced" />
+
+  <!-- TODO: Replace HelpUrl with a fwlink -->
+  <!-- TODO: Validation -->
+  <StringProperty Name="BaseAddress"
+                  DisplayName="DLL base address"
+                  HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/advanced-compiler-settings-dialog-box-visual-basic?view=#optimizations"
+                  Category="Advanced" />
+
+  <!-- TODO: Replace HelpUrl with a fwlink -->
+  <EnumProperty Name="DebugType"
+                DisplayName="Debug symbols"
+                Description="Specifies the kind of debug symbols produced during build."
+                HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/advanced-compiler-settings-dialog-box-visual-basic?view=#optimizations"
+                Category="Advanced">
+    <EnumProperty.Metadata>
+      <NameValuePair Name="SearchTerms" Value="debug type" />
+    </EnumProperty.Metadata>
+    <EnumValue Name="none"
+               DisplayName="No symbols are emitted" />
+    <!--
+    Note that 'pdbonly' is the same as 'full'.
+    <EnumValue Name="pdbonly"
+               DisplayName="PDB Only" />
+    -->
+    <EnumValue Name="full"
+               DisplayName="PDB file, current platform" />
+    <EnumValue Name="portable"
+               DisplayName="PDB file, portable across platforms" />
+    <EnumValue Name="embedded"
+               DisplayName="Embedded in DLL/EXE, portable across platforms" />
+  </EnumProperty>
+
+  <!-- TODO: Replace HelpUrl with a fwlink -->
+  <BoolProperty Name="DefineDebug"
+                DisplayName="Define DEBUG constant"
+                HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/advanced-compiler-settings-dialog-box-visual-basic#compilation-constants"
+                Category="Advanced"/>
+
+  <!-- TODO: Replace HelpUrl with a fwlink -->
+  <BoolProperty Name="DefineTrace"
+                DisplayName="Degine TRACE constant"
+                HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/advanced-compiler-settings-dialog-box-visual-basic#compilation-constants"
+                Category="Advanced"/>
+
+  <!-- TODO: Replace HelpUrl with a fwlink -->
+  <StringProperty Name="DefineConstants"
+                  DisplayName="Custom constants"
+                  HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/advanced-compiler-settings-dialog-box-visual-basic#compilation-constants"
+                  Category="Advanced">
+    <StringProperty.Description>Enter any custom constants for your application in this text box. Entries should be delimited by commas, using this form: Name1="Value1",Name2="Value2",Name3="Value3".</StringProperty.Description>
```

Comment:
Could we re-use the `NameValueList` editor here? The format looks a little different, so perhaps we'd need to add the encoding extension point that we've discussed previously.

Reply:
I would certainly like to, but as this has _always_ been a simple string that depended on proper user formatting it's a lower priority.

Reply:
Is the encoding actually different? If not, it should be very cheap to change the editor here.